src/main/java/frc/team4909/team4909SwerveModule.java

```diff
@@ -0,0 +1,51 @@
+package frc.team4909;
+
+import frc.bionic.swerve.*;
+
+public class team4909SwerveModule extends frc.bionic.swerve.SwerveModule{
+
+  /**
+  Gear ratio of first two pairs of gears.
+  Yaw does not depend on the third pair.
+  */
+
+  //TODO Get values
+  private static final double       GEAR_RATIO_YAW          =  0; 
+
+  // Gear ratio of all three pairs of gears.
+  // Wheel speed depends on all three.
+  private static final double       GEAR_RATIO_WHEEL_SPEED  =  0; 
+
+  // Maximum yaw speed in RPM
+  private static final double       MAX_YAW_SPEED_RPM       = 0; 
+
+  // The wheel diameter, in meters
+  private static final double      WHEEL_DIAMETER_METERS    = 0; 
+
+  public team4909SwerveModule(int canDeviceChannelA, int canDeviceChannelB,
+                            int dioEncoderChannelA, int dioEncoderChannelB,
+                            String name, String shuffleboardTabName)
+  {
+    super(name, shuffleboardTabName);
+
+    // Instantiate the two swerve motors our swerve module incorporates
+    IMotor motorA = new MotorFaclon500(canDeviceChannelA, name + " A", shuffleboardTabName);
+    IMotor motorB = new MotorFaclon500(canDeviceChannelB, name + " B", shuffleboardTabName);
+
+    // Instantiate the yaw encoder our swerve module incorporates
+    IYawEncoder encoder;
+    if (name == "RR")   // currently, right-rear has RevHex encoder; others have Grayhill
+    {
+      encoder = new YawEncoderRevHex(dioEncoderChannelA, name, shuffleboardTabName);
```

Comment:
We have the RevHex encoders working well, and Grayhill was being difficult. I think you could get rid of the Grayhill `else` condition, leaving only the RevHex regardless of `name`. We could also, therefore, change the two parameters, `dioEncoderChannelA` and `dioEncoderChannelB` to a single parameter, `dioEncoderChannel`.